server/message_queue_service.py

```diff
@@ -49,29 +61,30 @@ async def _connect(self) -> None:
                 loop=asyncio.get_running_loop(),
             )
         except ConnectionError:
-            self._logger.warning("Unable to connect to RabbitMQ. Is it running?", exc_info=True)
-            return
+            self._logger.warning(
+                "Unable to connect to RabbitMQ. Is it running?", exc_info=True
+            )
+            raise ConnectionAttemptFailed
         except ProbableAuthenticationError:
             self._logger.warning(
-                "Unable to connect to RabbitMQ. Incorrect credentials?",
-                exc_info=True
+                "Unable to connect to RabbitMQ. Incorrect credentials?", exc_info=True
             )
-            return
+            raise ConnectionAttemptFailed
```

Comment:
Maybe use the `raise from` syntax.

```suggestion
            raise ConnectionAttemptFailed from e
```
(need to change the `except` block to name the exception)